src/rust/ensogl/lib/components/src/selector/common/shape.rs

```diff
@@ -0,0 +1,174 @@
+use crate::prelude::*;
+
+use ensogl_core::display::shape::*;
+use ensogl_theme as theme;
+
+use crate::shadow;
+
+
+
+// ==================
+// === Background ===
+// ==================
+
+/// Utility struct that contains the background shape for the selector components, as well as some
+/// meta information about it. This information can be used to align other shapes with the
+/// background.
+#[allow(dead_code)]
```

Comment:
Not all the fields are used, but I left them for completeness. It feels wrong to just have one of width/height, for example. 
Instead of the `#[allow(dead_code)]` I made the fields public, do better reflect intent: the fields are there to be used if needed.